hledger-lib/Hledger/Data/Dates.hs

```diff
@@ -292,6 +293,30 @@ earliest (Just d1) (Just d2) = Just $ min d1 d2
 daysSpan :: [Day] -> DateSpan
 daysSpan ds = DateSpan (minimumMay ds) (addDays 1 <$> maximumMay ds)
 
+-- | Calculate the DateSpan containing a given Day, if any, from a given list of
+-- DateSpans.
```

Comment:
Select instead of Calculate, perhaps ?

How about calling it latestSpanContaining ?

Reply:
Again, sounds reasonable.